message/router/middleware/duplicator.go

```diff
@@ -0,0 +1,22 @@
+package middleware
+
+import (
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// Duplicator sends message twice.
+func Duplicator(h message.HandlerFunc) message.HandlerFunc {
+	return func(msg *message.Message) ([]*message.Message, error) {
+		producedMessages, firstErr := h(msg)
+		if firstErr != nil {
+			return nil, firstErr
+		}
+
+		_, secondErr := h(msg)
```

Comment:
I was thinking about it for a while, and probably we want to append it to `producedMessages`, eventually we can add config for this decorator (like in `Retry`) or separated decorator (like in `PoisonQueue` or `PoisonQueueWithFilter`).

Reply:
Fix c8baf8aad31ef594e153cc16f03f682864df26c4